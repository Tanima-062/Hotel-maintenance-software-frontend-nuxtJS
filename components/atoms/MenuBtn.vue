<template>
  <button
    type="button"
    class="flex justify-center items-center leading-5 whitespace-no-wrap"
    :class="styleColor"
    @click="click"
    @mouseover="mouseover"
    @mouseleave="mouseleave"
  >
    <slot />
  </button>
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: 'text-black' })
  textColor!: string

  @Prop({ default: false })
  isActive!: boolean

  private isAnim: boolean = false
  private interval: number | undefined

  get styleColor(): string {
    const classList = []
    // click時のボタン
    classList.push(this.isPushed ? 'btn--click' : 'btn')

    // テキスト文字の切り替え
    classList.push(this.textColor)

    // 背景色の切り替え
    classList.push(this.isPushed ? 'bg-menuColor-hover font-bold' : '')

    return classList.join(' ')
  }

  get isPushed(): boolean {
    return this.isAnim || this.isActive
  }

  private doAnim(): void {
    if (this.isAnim || this.interval) {
      return
    }
    this.isAnim = true
    this.interval = window.setInterval(() => {
      this.isAnim = false
      clearInterval(this.interval)
      this.interval = undefined
    }, 150)
  }

  @Emit()
  private click(e: Event): Event {
    this.doAnim()
    return e
  }

  @Emit()
  mouseover(e: Event): Event {
    return e
  }

  @Emit()
  mouseleave(e: Event): Event {
    return e
  }
}
</script>
<style lang="sass" scoped>
.btn
  transition: background-color .3s ease-in
  &:hover
    transition: opacity .1s cubic-bezier(.4,0,.6,1)
    opacity:.9
  &--click
    transition: background-color .15s ease
</style>
