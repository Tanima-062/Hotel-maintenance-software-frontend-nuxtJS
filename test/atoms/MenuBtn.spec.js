import { shallowMount } from '@vue/test-utils'
import MenuBtn from '@/components/atoms/MenuBtn.vue'

describe('Atoms MenuBtn Check', () => {
  test('MenuBtn Test', () => {
    const wrapper = shallowMount(MenuBtn)

    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
