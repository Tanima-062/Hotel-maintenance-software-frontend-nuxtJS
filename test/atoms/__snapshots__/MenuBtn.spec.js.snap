// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms MenuBtn Check MenuBtn Test 1`] = `
<button
  class="flex justify-center items-center leading-5 whitespace-no-wrap btn text-black "
  type="button"
/>
`;
